### ISSUE TITLE
What is difference between ExMatch and Snorkel results?

### ISSUE THREAD

**Issue by 009deep** [yueyu1030/COSINE#12]
Hi, I have a question about the results shared in the table of [paper](https://arxiv.org/pdf/2010.07835.pdf).

What is the difference between ExMatch and the Snorkel model? Were there different rule sets used in snorkel to generate a weak label and then its  label_model was applied to weak labels from different rule sets?

**Comment by yueyu1030** [comment]
ExMatch is the result of the majority voting method. We do not use probabilistic model (i.e. snorkel) to aggregate over different weak supervision sources.

**Comment by 009deep** [comment]
I see. Thanks for the clarification.

Also, what is the difference between _Init_ and _RoBERTa-WL_ . I  didn't actually get what `Init` is?  Is it like  _RoBERTa-WL_  is trained for more epochs using weak labels and thus suffers accuracy due to weak labels? And is `Init` results of _trainer.train_ step before applying  _trainer.selftrain_

**Comment by 009deep** [comment]
Actually, I managed to train the model on different public datasets with the approach. Closing it. It's really impressive. Thank s for  sharing details.

**Comment by yueyu1030** [comment]
> I see. Thanks for the clarification @yueyu1030
> 
> I have few more questions: what is the difference between _Init_ and _RoBERTa-WL_ . I didn't actually get what `Init` is? Is it like _RoBERTa-WL_ is trained for more epochs using weak labels and thus suffers accuracy due to weak labels? And is `Init` results of _trainer.train_ step before applying _trainer.selftrain_
> 
> Also, if I understood the code correctly, I'll talk specifically for `run_agnews.sh`. Here T1 is 150 and batch size is 32. Does it mean you fine-tune roberta model using 4800 (150*32) samples out of a total of ~55K training samples?

Your understanding is correct! Here we would like to highlight that if we use the epoch-wise evaluation method as roberta-wl does, then the model will overfit to the label noise even after the 1st epoch.

**Comment by yueyu1030** [comment]
Nice to hear that and feel free to let me know if there are other questions :)
